src/sync/discovery.ts

```diff
@@ -3,22 +3,54 @@ import { getRepositorySummary } from "./jobs";
 import enhanceOctokit from "../config/enhance-octokit";
 import { Application } from "probot";
 import { Repositories, SyncStatus } from "../models/subscription";
-import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import { LoggerWithTarget } from "probot/lib/wrap-logger";
 import { sqsQueues } from "../sqs/queues";
+import GitHubClient from "../github/client/github-client";
+import { getCloudInstallationId } from "../github/client/installation-id";
+import { Repository } from "@octokit/graphql-schema";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
 
 export const DISCOVERY_LOGGER_NAME = "sync.discovery";
+const MAX_PAGE_SIZE_REPOSITORY = 100;
+
+const getAllRepositories = async (github, hasNextPage: boolean, cursor?: string, repositories: Repository[] = []): Promise<Repository[]> => {
+	if (!hasNextPage) {
+		return repositories;
+	}
+
+	const result = await github.getRepositoriesPage(MAX_PAGE_SIZE_REPOSITORY, cursor);
+	const edges = result?.viewer?.repositories?.edges || [];
+	const nodes = edges.map(({ node: item }) => item);
+	const repos = [...repositories, ...nodes];
```

Comment:
Aren't we risking OOM if we have to keep the total list of repositories fetched on each iteration?

Reply:
That won't be garbage collected until recursion finished.